### ISSUE TITLE
In promoting jianpu-ly

### ISSUE THREAD

**Issue by suntong** [ssb22/jianpu-ly#26]
Hi @ssb22, 

I had been thinking of different ways I can help promoting the wider-adoption of jianpu-ly, and here are my two thoughts:

- Grant me access to the wiki, so that I can put many of my closed questions there, as they would be common for any jianpu-ly starters. 

- I've also collected all possible jianpu-ly based music files I could find on the internet into a github repo, would you think if it is a good idea that we move both your jianpu-ly software and my jianpu-ly based music file collection under a single github organization, so that it'll make jianpu-ly newcomers to get easier start? (If so, please accept the github organization invitation as I've done some legworks for you)

**Comment by ssb22** [comment]
Thanks.  I'm not 100% sure how GitHub organisations & wikis work, but we can look at it.  Moving a project worries me: does it mean the URL will change, and will everybody who's already `git clone`d from the old URL still be able to `git pull`?  If not, a move could be disruptive and I wonder if the organisation could just link back to the original repo instead....

**Comment by suntong** [comment]
Moving a project does change its url (and git clone/pull url too), however, if you don't create another repo with the same name, then people who's already `git cloned` from the old URL can still be able to `git pull`, as GH will do the redirect automatically. You can try creating a small test repo under github.com/ssb22 then moved it to under jianpu-ly, then try doing the `git pull` and `git push`. 

If you feel comfortable about it, then we can talk about moving the main repo over, just that its url does get changed. Give it a try and then we can decide...

BTW, the wiki is working for me, and I've already put into one TIP note. 

thanks

**Comment by ssb22** [comment]
Testing with https://github.com/jianpu-ly/jianpu-test - see readme there - old URLs work, but old list of "my repositories" no longer lists it, which could be a minor hassle.  If I can fix [git2gmi](https://github.com/ssb22/bits-and-bobs/blob/master/git2gmi.py) then I might at least be able to use that.  Or maybe I'm just navigating GitHub wrongly....

**Comment by suntong** [comment]
Yep, I checked the items you put in https://github.com/jianpu-ly/jianpu-test, and apparently the impact is more than I anticipated. 

So let's park this idea for now until another person would like to contribute into jianpu-ly collection, so that we can _"group multiple repositories under the project"_ by then.

**Comment by suntong** [comment]
> So let's park this idea for now until ...

Ah, I now have a perfect solution -- mirror, instead of move!

I know you have a few more git repos than this, and when you push, you push them altogether. So, 

- Fork this repo into/under jianpu-ly
- Add the forked repo under jianpu-ly to your pushed remotes
- Make a small change, like doc [KeepOctave & KeepLength commands](https://github.com/ssb22/jianpu-ly/commit/1680493810cac498d29aa4cea1769577a6b838ac) in readme, then push, and make sure the forked repo under jianpu-ly is pushed as well. 

If AOK, add me as the admin to the forked repo under jianpu-ly, I'll co-maintain it with you.

**Comment by ssb22** [comment]
This could work, although apparently it's [difficult to sync issues and pull requests between forks](https://stackoverflow.com/questions/69394819/original-repository-issues-on-forked-repository)—don’t know if that’s important.

There are a few mirrors already: when GitHub was taken over by Microsoft, I got nervous and set up mirrors of all my public repositories on other platforms “just in case”.  The nice thing about Git is you *can* do that—it’s just a matter of adding extra push URLs to my local copy of the repo, which I [scripted](https://github.com/ssb22/bits-and-bobs/blob/master/repo-setup.sh)—so we’re not tied to one platform.  (That's why I always write commit messages saying "#26 on GitHub" instead of just "#26".)  Only trouble is, the mirrors share the Git history but they don't share the issues history.

**Comment by suntong** [comment]
> apparently it's [difficult to sync issues and pull requests between forks](https://stackoverflow.com/questions/69394819/original-repository-issues-on-forked-repository)

Yep, that's actually my next question -- we cannot keep both so we need to stick to one place for  issues and pull requests. If you want to keep them here then it is simple, I can turn off the issues and pull requests feature at the fork (and make a note in its wiki) so people will not be confused.

**Comment by suntong** [comment]
Any progress, blockage, or concerns?

**Comment by ssb22** [comment]
Sorry I've not yet tried this.  I need to figure out how to fork my own repository.  I can create a new repository (not a fork) and push to both, but that would make the relationship less obvious.  It's not clear how to create a fork if I myself am the owner of the repo being forked; I'm not sure if GitHub even allows this.  Maybe you'll have to be the one to create the fork.

**Comment by suntong** [comment]
> Maybe you'll have to be the one to create the fork.

Oh, sure, I didn't realized that this could be an issue. Glad that I can help.

**Comment by suntong** [comment]
Done.

I've turned off the issues and added a note not to do pull requests.

All set, and all your now.

**Comment by ssb22** [comment]
Thanks.  OK seeing as there's a new commit I'm going to see what happens if I try to `git push` directly to the new fork....

**Comment by ssb22** [comment]
... it doesn't work :-(

    git remote set-url origin --push --add git@github.com:jianpu-ly/jianpu-ly.git
    git push
    ...
    ERROR: Permission to jianpu-ly/jianpu-ly.git denied to ssb22.

Maybe it doesn't automatically give me permission just because I'm in the same organisation, and you have to explicitly set it on every repository you create.

**Comment by suntong** [comment]
Aha, you're right. (I'm doing this for the first time as well)

Try again pls.

**Comment by ssb22** [comment]
Thanks, it worked that time, and the GitHub Web interface correctly recognised it as the same commit and now says "This branch is up to date with ssb22/jianpu-ly:master" (not "1 commit ahead, 1 commit behind" as it would have said if it hadn't recognised it as the same commit).  Good.

Interestingly, when I did the `git push --tags`, it said all version tags right back to v1.145 (when I started doing version tags) were pushed to the fork.  That suggests creating a fork on GitHub does not by default copy the version tags, which is an oversight on their part.  Not to worry though if we're going to be pushing them anyway.
